### PR TITLE
core: refactor pfcq to follow single-point exit approach

### PR DESCRIPTION
**Pull request by MathWay**

### CODE REVIEW

File: pfcq.c [hunk 1]
```diff
@@ -121,16 +121,16 @@ void __pfcq_fail(const char* _message, const int _errno)
 	return;
 }
 
-void __pfcq_stop(const char* _message)
+void __pfcq_stop(const int _exit_code, const char* _message)
 {
 	__pfcq_debug(1, "%s\n", _message);
```
Comment (pfactum) [inline pfcq.c]: May we have check for empty string here not to put extra \n if "" is passed?

**Comment by MathWay** [comment]
Please, check.

File: pfcq.c [hunk 2]
```diff
@@ -121,16 +121,17 @@ void __pfcq_fail(const char* _message, const int _errno)
 	return;
 }
 
-void __pfcq_stop(const char* _message)
+void __pfcq_stop(const int _exit_code, const char* _message)
 {
-	__pfcq_debug(1, "%s\n", _message);
-	exit(EX_SOFTWARE);
+	if (unlikely(!_message))
```
Comment (pfactum) [inline pfcq.c]: We have missed the whole forest behind trees here.

__pfcq_debug() should be called only and if only _message is not NULL:

`if (likely(_message))
    __pfcq_debug();`

**Comment by MathWay** [comment]
Fixed.

**Pull request merged**